src/webui/WebUI.jl

```diff
@@ -0,0 +1,227 @@
+
+const DOCS = "https://github.com/JuliaRegistries/Registrator.jl/blob/master/README.web.md#usage-for-package-maintainers"
+const CONFIG = Dict{String, Any}()
+
+include("../management.jl")
+const httpsock = Ref{Sockets.TCPServer}()
+
+include("providers.jl")
+
+# The target registry.
+struct Registry{F <: GitForge.Forge, R}
+    forge::F
+    repo::R
+    url::String
+    clone::String
+    deps::Vector{String}
+    enable_patch_notes::Bool
+end
+
+# U is a User type, e.g. GitHub.User.
+struct User{U, F <: GitForge.Forge}
+    user::U
+    forge::F
+end
+
+struct RegistrationData
+    project::Pkg.Types.Project
+    tree::AbstractString
+    repo::Union{GitForge.GitHub.Repo, GitForge.GitLab.Project}
+    user::Union{GitForge.GitHub.User, GitForge.GitLab.User}
+    ref::AbstractString
+    commit::AbstractString
+    notes::AbstractString
+end
+
+const event_queue = Channel{RegistrationData}(1024)
+
+struct RegistrationState
+    msg::AbstractString
+    state::Symbol
+end
+
+const PROVIDERS = Dict{String, Provider}()
+const REGISTRY = Ref{Registry}()
+const USERS = TTL{String, User}(Hour(1))
+const REGISTRATIONS = Dict{AbstractString, RegistrationState}()
+
+###########
+# Helpers #
+###########
+
+include("webutils.jl")
+include("gitutils.jl")
+
+##########
+# Routes #
+##########
+
+include("routes/index.jl")
+include("routes/auth.jl")
+include("routes/callback.jl")
+include("routes/select.jl")
+include("routes/register.jl")
+include("routes/status.jl")
+
+##############
+# Entrypoint #
+##############
+
+function init_registry()
+    url = CONFIG["registry_url"]
+    k = get(CONFIG, "registry_provider") do
+        if occursin("github", url)
+            "github"
+        elseif occursin("gitlab", url)
+            "gitlab"
+        end
+    end
+    haskey(PROVIDERS, k) || error("Unsupported registry host")
+    forge = PROVIDERS[k].client
+    owner, name = splitrepo(url)
+    repo = @gf get_repo(forge, owner, name)
+    repo === nothing && error("Registry lookup failed")
+    clone = get(CONFIG, "registry_clone_url", url)
+    deps = get(CONFIG, "registry_deps", String[])
+    enable_patch_notes = !get(CONFIG, "disable_patch_notes", false)
+    REGISTRY[] = Registry(forge, repo, url, clone, deps, enable_patch_notes)
+end
+
+for f in [:index, :auth, :callback, :select, :register]
+    @eval $f(f::Function, r::HTTP.Request) = f($f(r))
+end
+
+error_handler(f::Function, r::HTTP.Request) = try
+    f(r)
+catch e
+    println(get_backtrace(e))
+    @error "Handler error" route=r.target
+    html(500, "Server error, sorry!")
+end
+
+pathmatch(p::AbstractString, f::Function) = branch(r -> first(split(r.target, "?")) == p, f)
+
+function action(regdata::RegistrationData, zsock::RequestSocket)
+    regp = RegEdit.RegisterParams(cloneurl(regdata.repo), regdata.project, regdata.tree;
+        registry=REGISTRY[].clone, registry_deps=REGISTRY[].deps, push=true,
+        gitconfig=Dict("user.name"=>CONFIG["user"], "user.email"=>CONFIG["email"]),
```

Comment:
```suggestion
        gitconfig=Dict("user.name" => CONFIG["user"], "user.email" => CONFIG["email"]),
```

Reply:
Also, in my opinion these values should not be required, since global Git configuration can be used instead.
Perhaps:
```julia
gitconfig = Dict{String, String}()
haskey(CONFIG, "user") && (gitconfig["user.name"] = CONFIG["user"])
haskey(CONFIG, "email") && (gitconfig["user.email"] = CONFIG["email"])
```